src/io_uring.c

```diff
@@ -0,0 +1,92 @@
+#include "io_uring.h"
+
+#ifdef HAVE_LIBURING
+#include <liburing.h>
+
+#define IO_URING_DEPTH 256 /* io_uring instance queue depth. */
+
+static size_t io_uring_queue_len = 0;
+
+void initIOUring(void) {
+    if (server.io_uring_enabled) {
+        struct io_uring_params params;
+        struct io_uring *ring = zmalloc(sizeof(struct io_uring));
+        memset(&params, 0, sizeof(params));
+        /* On success, io_uring_queue_init_params(3) returns 0 and ring will
+         * point to the shared memory containing the io_uring queues.
+         * On failure -errno is returned. */
+        int ret = io_uring_queue_init_params(IO_URING_DEPTH, ring, &params);
+        if (ret != 0) {
+            /* Warning if user enable the io_uring in config but system doesn't support yet. */
+            serverLog(LL_WARNING, "Failed to initialize io_uring: %s", strerror(-ret));
+            zfree(ring);
+            server.io_uring = NULL;
```

Comment:
shouldn't we exit here? the config says io_uring should be enabled but we can't use io_uring. Continuing in this case violates the contract IMO.
```suggestion
        serverAsssert(io_uring_queue_init_params(IO_URING_DEPTH, ring, &params) == 0);
```

Reply:
The intention is to fall back the non io_uring if running environment didn't support io_uring for some reasons like kernel version too old etc.  
But I am fine to failed quickly now if user manually enable io_uring, maybe later when io_uring is enabled default we can keep this log.

Reply:
How do we do for other configs? Like TLS if OpenSSL is not installed?

Reply:
For TLS. Log and exit.
``` c
    if (server.tls_port || server.tls_replication || server.tls_cluster) {
        ConnectionType *ct_tls = connectionTypeTls();
        if (!ct_tls) {
            serverLog(LL_WARNING, "Failed finding TLS support.");
            exit(1);
        }
        if (connTypeConfigure(ct_tls, &server.tls_ctx_config, 1) == C_ERR) {
            serverLog(LL_WARNING, "Failed to configure TLS. Check logs for more info.");
            exit(1);
        }
    }
```

Reply:
OK, so maybe "log and exit" is better also for io_uring? I'm not sure.

Reply:
Ok, keep alignment with TLS, log and exit.